src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -384,12 +377,40 @@ def _get_losses(self, outputs: torch.Tensor, targets: torch.Tensor) -> Tuple[tor
         else:
             loss_logging_items = loss.unsqueeze(0).detach()
 
+        # ON FIRST BACKWARD, DERRIVE THE LOGGING TITLES.
+        if self.loss_logging_items_names is None:
+            self._init_loss_logging_names(loss_logging_items)
+            self._init_monitored_items()
```

Comment:
Sorry to interrupt, I think you have a bug here.
In `self._init_loss_logging_names` you have a row with:
```python
elif len(loss_logging_items) > 1
```
which will raise `TypeError: object of type 'NoneType' has no len()`